gnomad_qc/v5/annotations/generate_frequency.py

```diff
@@ -0,0 +1,1171 @@
+    logger.info("Applying v4 genomes hom alt depletion fix...")
+    vmt = vmt.annotate_entries(
+        GT=hom_alt_depletion_fix(
+            vmt.GT,
+            het_non_ref_expr=vmt._het_non_ref,
+            af_expr=vmt.v4_af,
+            ab_expr=vmt.AD[1] / vmt.DP,
+            use_v3_1_correction=True,
+        )
+    )
+    logger.info("Annotating age distribution...")
+    vmt = vmt.annotate_globals(
+        age_distribution=vmt.aggregate_cols(
+            hl.agg.hist(vmt.age, AGE_HIST_MIN, AGE_HIST_MAX, AGE_HIST_BINS)
+        )
+    )
+
+    vds = hl.vds.VariantDataset(vds.reference_data, vmt)
+    return vds.checkpoint(new_temp_file("consent_samples_vds_prepared", "vds"))
+
+
+def _calculate_consent_frequencies_and_age_histograms(
+    vds: hl.vds.VariantDataset,
+    test: bool = False,
+) -> hl.Table:
+    """
+    Calculate frequencies and age histograms for consent withdrawal samples.
+
+    :param vds: Prepared VDS with consent samples.
+    :param test: Whether running in test mode.
+    :return: Table with freq and age_hists annotations for consent samples.
+    """
+    logger.info("Densifying VDS for frequency calculations...")
+    mt = hl.vds.to_dense_mt(vds)
+    consent_sample_ids = set(mt.s.collect())
+    group_membership_ht = group_membership(test=test, data_set="gnomad").ht()
+
+    logger.info(
+        "Filtering group membership table to %s consent samples...",
+        len(consent_sample_ids),
+    )
+    group_membership_ht = group_membership_ht.filter(
+        hl.literal(list(consent_sample_ids)).contains(group_membership_ht.s)
+    )
+    logger.info(
+        "Group membership table has been filtered to %s consent samples...",
+        group_membership_ht.count(),
+    )
+
+    mt = mt.annotate_cols(
+        group_membership=group_membership_ht[mt.col_key].group_membership,
+    )
+    mt = mt.annotate_globals(
+        freq_meta=group_membership_ht.index_globals().freq_meta,
+        freq_meta_sample_count=group_membership_ht.index_globals().freq_meta_sample_count,
+    )
+
+    logger.info(
+        "Calculating frequencies and age histograms using compute_freq_by_strata..."
+    )
+
+    # Annotate hists_fields on MatrixTable rows before calling compute_freq_by_strata so
+    # to keep the age_hists annotation on the frequency table.
+    logger.info("Annotating hists_fields on MatrixTable rows...")
+    mt = mt.annotate_rows(
+        hists_fields=hl.struct(
+            age_hists=age_hists_expr(mt.adj, mt.GT, mt.age),
+        )
+    )
+
+    logger.info(
+        "Computing frequencies for consent samples using compute_freq_by_strata..."
+    )
+    consent_freq_ht = compute_freq_by_strata(
+        mt,
+        select_fields=["hists_fields"],
+    )
+
+    consent_freq_ht = consent_freq_ht.transmute(
+        age_hists=consent_freq_ht.hists_fields.age_hists,
+    )
+
+    return consent_freq_ht.checkpoint(new_temp_file("consent_freq_and_hists", "ht"))
```

Comment:
i ran hl.eval(consent_freq_ht.freq_meta_sample_count)

and got 
[76215,
 76215,
 20805,
 456,
 7657,
 1736,...
 
 shouldn't the max value in the array be 866?